### PR TITLE
docs: Incremental improvements

### PR DESCRIPTION
**Pull request by mikemckiernan**
Partial completion of the items cataloged
in issue #180:

* Def and objective of the library
* Core features -- need Eng to enhance this
* Installation (minus dev env)
* Move the retrieval model info to sep page.

### CODE REVIEW

**Comment by marcromeyn** [comment]
Is this ready for review? If so, it could help to add the tag: `status/needs-review`.

File: README.md [hunk 1]
```diff
@@ -2,166 +2,72 @@
...
-Wang, Ruoxi, Rakesh Shivanna, Derek Z. Cheng, Sagar Jain, Dong Lin, Lichan Hong, and Ed H. Chi. “DCN V2: Improved Deep &
-Cross Network and Practical Lessons for Web-Scale Learning to Rank Systems.” ArXiv:2008.13535 [Cs, Stat], October 20, 2020. http://arxiv.org/abs/2008.13535.
+To learn about the core features of Merlin Models, see the [Retrieval Models](docs/source/retrieval_models.md) page.
+The key retrieval models are as follows:
 
-![img.png](docs/img/dcn-v2.png)
+* Matrix factorization
+* YouTube DNN retrieval
+* Two tower
+* Ranking models such as DLRM and DCN-V2
```
Comment (marcromeyn) [inline README.md:55]: These are `Ranking Models`. For some background for this, could be good to watch [this](https://www.youtube.com/watch?v=5qjiY-kLwFY) talk from Even.

Comment (mikemckiernan) [reply README.md:55]: Good video.  At this point though, I'm not convinced that it's helpful to treat "ranking models" as a proper noun.  There's also some precedent for using plain lower-case text for other model types in the terminology repository that the docs team uses.

**Review by marcromeyn — APPROVED** [review]
Looks good to me!

**Review by benfred — APPROVED** [review]

**Pull request merged**